### DISCUSSION TITLE
[Proposal] Operator Constraint on Generic Type

### DISCUSSION THREAD

**Discussion by dangi12012** [dotnet/csharplang#1532] [Language Ideas] [answered]
Maybe consider adding contraints for operators and casts? **Since operators cannot be added in an interface I cannot add 2 generic Variables together**. Would be cool for a generic math library where we dont want a class for every single valuetype. 
Missing these constraints was a major reason in adding dynamic keyword but by adding these constraints we get MUCH speed back. (Which is a good thing)
 
where T : Primitive (int, byte, long, etc. so we know all interfaces the common number types have)
where T : Castable to T1 (We can do ((T1)T).Member)
where T : Operator+(T1, T2) //When the result of an operator is not T (Matrix * Vector)
where T : Operator+ //When the result of an operator is T (int + int)

**Comment by HaloFour** [comment]
#110

**Comment by yaakov-h** [comment, marked as answer]
#164 too, I think.
